### PR TITLE
Backport/60x/filemagic/min/v4

### PR DESCRIPTION
**Pull request by victorjulien**
#9351 with minor cleanups.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1346

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 15526

File: src/detect-filename.c [hunk 1]
```diff
@@ -142,10 +142,19 @@ void DetectFilenameRegister(void)
 
     g_file_match_list_id = DetectBufferTypeGetByName("files");
 
-    AppProto protos_ts[] = { ALPROTO_HTTP, ALPROTO_SMTP, ALPROTO_FTP, ALPROTO_FTPDATA, ALPROTO_SMB,
-        ALPROTO_NFS, 0 };
-    AppProto protos_tc[] = { ALPROTO_HTTP, ALPROTO_FTP, ALPROTO_FTPDATA, ALPROTO_SMB, ALPROTO_NFS,
```
Comment (jufajardini) [inline src/detect-filename.c:144]: Question: isn't this supported by ˋHTTP2ˋ? The documentation makes me expect it to be.

**Review by jufajardini — COMMENTED** [review]
Changes seem to make sense to me, left some inline comments where I got some doubts, but mainly nothing that blocks these changes.

File: src/detect-filename.c [hunk 2]
```diff
@@ -517,67 +382,11 @@ static int DetectFilenameSignatureParseTest01(void)
     //FAIL_IF_NOT(UTHParseSignature("alert tls any any -> any any (flow:to_client; file.name; content:\"abc\"; sid:1;)", false));
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply src/detect-filename.c:430]: in master maybe, but not for backports

File: src/detect-filename.c (hunk 1 above)
Comment (victorjulien) [reply src/detect-filename.c:144]: Possibly, maybe. We'd need to check why this isn't done in 6. @catenacyber ?

**Comment by victorjulien** [comment]
Replaced by #9358

**Pull request closed without merging**